### PR TITLE
docs: add "concepts" page

### PR DESCRIPTION
**Pull request by aaronsteers**
This page attempts to clarify specific concepts, internally and externally, and to provide a shared language around these core concepts.

### CODE REVIEW

File: docs/concepts.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+Each Cloud Environment Deployment must specify the environment name to deploy and the git branch to use when tracking project updates.
+
+> **Note**
+>
+> All operations that perform compute require a deployed environment - including: ad-hoc job execution, EL pipelines, scheduled tasks, etc.
+
+## Meltano Cloud Schedules
+
+Schedules within Meltano Cloud map directly to schedules defined in `meltano.yml`. However, in Meltano Cloud, each schedule starts off disabled by default, and should be enabled and disabled on a per-environment basis.
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply docs/concepts.md]: Updated to be more explicit. 👍

File: docs/concepts.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+Your Meltano Cloud Organization is the top-level entity associated with your payment info. Each organization will be issued a unique billing account number.
+
+> **Note**
+>
+> Each Meltano Cloud Organization can be mapped to zero or more GitHub organizations. See [Meltano Project Groups](#meltano-project-groups-and-github-orgs) below for more information.
+
+## Meltano Project Groups and "GitHub Orgs"
+
+Within each Meltano Cloud Organization, you can authorize Meltano Cloud to access your GitHub organizations. Internally, we refer to these as Meltano Cloud "project groups". Projects within a project group may share certain aspects, such as shared encryption/decryption keys and a shared set of role-based access permissions.
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply docs/concepts.md]: I removed all references to "Project Groups", instead adding a notes section that just clarifies that GitHub Orgs and Meltano Cloud Organizations are not 1:1.

[the remark this answers is not in the document]
Comment (WillDaSilva) [reply docs/concepts.md]: But we still plan to implement groups as a means of organization right? Will we update the docs once that feature has been implemented?

File: docs/concepts.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Meltano Cloud Concepts
+
+## Meltano Cloud Users
+
+Your Meltano Cloud user account is associated with a GitHub account for single-sign-on. If you do not have a GitHub account, you can create a free account at [github.com](https://github.com).
+
+> **Note**
+>
+> During the Private and Public Beta phases, we will be collecting feedback from users regarding the login and account creation experience overall. Post-GA, we plan to offer SAML as a premium feature for Enterprise users. If you have a use case to support alternate forms of login besides GitHub auth and SAML (e.g. GitLab, or user/password auth), please let your account manager know or log an issue in our [issue tracker](https://github.com/meltano/cloud-docs/issues).
```
Comment (tayloramurphy) [inline docs/concepts.md:9]:
```suggestion
> During the Private and Public Beta phases, we will be collecting feedback from users regarding the login and account creation experience overall. Post-GA, we plan to offer SAML on Cloud. If you have a use case to support alternate forms of login besides GitHub auth and SAML (e.g. GitLab, or user/password auth), please let your account manager know or log an issue in our [issue tracker](https://github.com/meltano/cloud-docs/issues).
```

File: docs/concepts.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+>
+> Meltano Cloud Organizations should not be confused with GitHub Organizations (or "GitHub Orgs"). Each Meltano Cloud Organization can include projects from multiple GitHub Orgs.
+>
+> Meltano Cloud users grant access to specific public and private repositories using the standard GitHub App authorization flow.
+
+## Meltano Cloud Projects
+
+A "Cloud Project" is any Meltano project you have registered on Meltano Cloud. The project definition consists of a git repo and a relative project directory within the git repo.
+
+## Meltano Cloud Environments (aka "Deployments")
```
Comment (tayloramurphy) [inline docs/concepts.md:25]:
```suggestion
## Meltano Cloud Deployments
```

What do you think about just sticking with Deployment here? Mixing in environment again just feels clunky.

File: docs/concepts.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+Within each Cloud Project, you can deploy zero or more named [Meltano Environments](https://docs.meltano.com/concepts/environments) to Meltano Cloud. Internally, we often refer to these as "deployments" since they are isolated environments deployed to the Cloud.
+
+Each Cloud Environment Deployment must specify the environment name to deploy and the git branch to use when tracking project updates.
```
Comment (tayloramurphy) [inline docs/concepts.md:29]:
```suggestion
Each Cloud Deployment must specify the environment name to deploy and the git branch to use when tracking project updates.
```

File: docs/concepts.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+> **Note**
+>
+> All operations that perform compute require a deployed environment - including: ad-hoc job execution, EL pipelines, scheduled tasks, etc.
+
+## Meltano Cloud Schedules
+
+Schedules within Meltano Cloud map directly to schedules defined in `meltano.yml`. However, in Meltano Cloud, each schedule starts off disabled by default, and each schedule is enabled or disabled on a per-environment basis.
```
Comment (tayloramurphy) [inline docs/concepts.md:37]:
```suggestion
Schedules within Meltano Cloud map directly to schedules defined in `meltano.yml`. However, in Meltano Cloud, each schedule starts off disabled by default, and each schedule is enabled or disabled on a per-Deployment basis.
```

**Review by tayloramurphy — CHANGES_REQUESTED** [review]

File: docs/concepts.md (hunk 4 above)
Comment (WillDaSilva) [reply docs/concepts.md:25]: Maybe "Deployed Environments" would be ideal? It's fairly clear as to what this means both to a Meltano user and a newcomer.

Comment (tayloramurphy) [reply docs/concepts.md:25]: Maybe... I'm still hestiant though b/c it's not just a deployed environment - it's a deployed environment with a specific git branch/sha.

File: docs/concepts.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+## Meltano Cloud Users
+
+Your Meltano Cloud user account is associated with a GitHub account for single-sign-on. If you do not have a GitHub account, you can create a free account at [github.com](https://github.com).
+
+> **Note**
+>
+> During the Private and Public Beta phases, we will be collecting feedback from users regarding the login and account creation experience overall. Post-GA, we plan to offer SAML as a premium feature for Enterprise users. If you have a use case to support alternate forms of login besides GitHub auth and SAML (e.g. GitLab, or user/password auth), please let your account manager know or log an issue in our [issue tracker](https://github.com/meltano/cloud-docs/issues).
+
+## Meltano Cloud Organizations
```
Comment (tayloramurphy) [inline docs/concepts.md:11]: The more I think about it, the more I like organizations. It's pretty clear what it represents and there's lots of prior art for it (https://cloud.google.com/resource-manager/docs/cloud-platform-resource-hierarchy)

File: docs/concepts.md (hunk 3 above)
Comment (aaronsteers) [reply docs/concepts.md:9]: 👍

File: docs/concepts.md (hunk 6 above)
Comment (aaronsteers) [reply docs/concepts.md:37]: 👍

File: docs/concepts.md (hunk 4 above)
Comment (aaronsteers) [reply docs/concepts.md:25]: I've updated the language here to use "Deployment" more consistently. Will commit and push momentarily.

File: docs/concepts.md (hunk 5 above)
Comment (aaronsteers) [reply docs/concepts.md:29]: ✅

**Comment by aaronsteers** [comment]
@tayloramurphy - Any last changes before I merge? We can still iterate from here.

**Review by tayloramurphy — APPROVED** [review]

**Pull request merged**